arbiter-core/src/environment.rs

```diff
@@ -32,63 +50,84 @@ pub enum State {
     Stopped,
 }
 
+/// The environment struct.
 pub struct Environment {
+    /// label for the environment
     pub label: String,
     pub(crate) state: State,
     pub(crate) evm: EVM<CacheDB<EmptyDB>>,
+    /// Connection to the environment
     pub connection: Connection,
     /// Clients (Agents) in the environment
-    pub clients: Vec<Arc<Agent<RevmMiddleware>>>,
-    // pub deployed_contracts: HashMap<String, Contract<RevmMiddleware>>,
+    pub clients: HashMap<String, Agent<RevmMiddleware>>,
```

Comment:
This shouldn't be necessary as all agents will be run in an environment. There's no need to reference by name.